darwin/future/core/types/query.py

```diff
@@ -56,6 +66,40 @@ def filter_attr(self, attr: Any) -> bool:  # type: ignore
         else:
             raise ValueError(f"Unknown modifier {self.modifier}")
 
+    @classmethod
+    def _from_dict(cls, d: Dict[str, Any]) -> QueryFilter:  # type: ignore
+        if "name" not in d or "param" not in d:
+            raise ValueError(f"args must be a QueryFilter or a dict with 'name' and 'param' keys, got {d}")
+        modifier = Modifier(d["modifier"]) if "modifier" in d else None
+        return QueryFilter(name=d["name"], param=str(d["param"]), modifier=modifier)
+
+    @classmethod
+    def _from_args(cls, *args: object, **kwargs: str) -> List[QueryFilter]:
```

Comment:
so `object` overcomes the mypy typing constraints as everything in python is an `object`, for mypy it's basically `Any` but when you actually mean "I accept anything" and not just "I cbf typing this input". 

Interestingly, while researching typing for mypy and args/kwargs functions, `object` and `Any` are weird, inherited objects of each other. It's a strange, one off relationship that python defines specifically. 

In terms of for this though, the intention was that I could provide an array, as well as a dict, but I actually just settled on checking for dict and didn't replace this typing. I could make it stricter and replace with 
```python
def _from_args(cls, *args: Dict[str, object], **kwargs: str) -> List[QueryFilter]

```

In your example though the `f=2` would cause a typing issue as I've defined only str. There's scope here to allow `object` instead and then cast to str for the QueryFilter and I'm not fussed, I think explicitly saying `str` here though does give better type hint about the intention as it's basically signalling that you need to be able to cast it as a str for the filtering object.

Reply:
https://stackoverflow.com/questions/39817081/typing-any-vs-object